tests/unit/test_migration_engine.py

```diff
@@ -0,0 +1,197 @@
+import random
+import sqlite3
+import string
+from sqlite3 import Cursor
+from typing import Any, Dict
+
+from pytest import raises
+
+from nxdrive.dao.migrations.migration import MigrationInterface
+from nxdrive.dao.migrations.migration_engine import MigrationEngine
+
+
+def generate_migration(
+    schema_version: int, previous_schema_version: int, /, *, should_raise: bool = False
+):
+    """Instantiate and return a migration object."""
+
+    class TestMigration(MigrationInterface):
+        table_name = "".join(random.choice(string.ascii_letters) for _ in range(5))
+
+        def __init__(self, schema_version, previous_schema_version) -> None:
+            self._schema_version = schema_version
+            self._previous_schema_version = previous_schema_version
+
+        def upgrade(self, cursor: Cursor) -> None:
+            """Create a table with a random name or generate an exception."""
+            if not should_raise:
+                cursor.execute(
+                    f"CREATE TABLE {self.table_name} ("
+                    "   path STRING NOT NULL,"
+                    "   PRIMARY KEY (path)"
+                    ")"
+                )
+                cursor.execute(f"PRAGMA user_version = {self.version}")
```

Comment:
Is it still needed since we moved such logic in the migration engine directly?

Reply:
you are right, it's not needed anymore